libvips/foreign/foreign.c

```diff
@@ -1548,6 +1548,7 @@ vips__foreign_convert_saveable(VipsImage *in, VipsImage **ready,
 	 */
 	if (in->Coding == VIPS_CODING_NONE &&
 		(saveable & VIPS_FOREIGN_SAVEABLE_ANY) &&
+		format &&
```

Comment:
This is a drive-by fix, but it might be redundant since `->format_table` already defaults to this:
https://github.com/libvips/libvips/blob/70b5eef9918110143f8fba8624311368d81e97ad/libvips/foreign/foreign.c#L1863-L1865

If `format` is guaranteed to never be `NULL`, we could simplify remove the `&& format` check on line 1643 instead.

Reply:
We also call `vips__foreign_convert_saveable()` directly from tiffsave.c and dzsave.c, though format is non-NULL there too, you're right. We could maybe add a `g_assert(format)` at the start of the function?

Reply:
Added an assert for both `format` and `coding` to be non-NULL and removed the redundant `&& format` check.